### DISCUSSION TITLE
Hot can I get the current Caret position of an AutoCompleteBox?

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [AvaloniaUI/Avalonia#13423] [Q&A] [answered]
I would like to make proposals for an AutoCompleteBox not only as extensions at the end. I want to make auto completetion proposals dependant on the current position of the caret. (a little bit like a wordwise auto completion)

That's why I want to know if there is a way to get the current caret position of an AutoCompleteBox? Or is there another way to figure out where the most recents change happend?

**Comment by timunie** [comment]
You can try VisualTreeExtensions to get TextBox child. There you have access to all releated Properties. 

If that's not enough, I'd implement my own control for it tbh. I have one for WPF  if you're interested. You need to figure out how to modify it to make it rin in Avalonia though.

**Reply by timunie** [reply]
Here is my WPF version. Again, it needs to be reworked by the user to make it run in Avalonia. 
Code:
https://github.com/timunie/TimsWpfControls/blob/master/TimsWpfControls/TimsWpfControls/Controls/IntellisenseTextBox.cs

Style: 
https://github.com/timunie/TimsWpfControls/blob/master/TimsWpfControls/TimsWpfControls/Themes/IntellisenseTextBox.xaml

**Reply by DevAM-Tools** [reply, marked as answer]
> You can try VisualTreeExtensions to get TextBox child. 

Thank you. The following snipped worked for me.

```
public partial class MainView : UserControl
{
    public MainView()
    {
        InitializeComponent();

        FilterAutoCompleteBox.TextChanged += FilterAutoCompleteBox_TextChanged;
    }

    private void FilterAutoCompleteBox_TextChanged(object sender, TextChangedEventArgs e)
    {
        if (DataContext == null)
        {
            return;
        }

        MainViewModel mainViewModel = (MainViewModel)DataContext;

        TextBox filterTextBox = (TextBox)FilterAutoCompleteBox.GetVisualDescendants().FirstOrDefault(x => x is TextBox);
        if (filterTextBox == null)
        {
            return;
        }

        mainViewModel.FilterCaretIndex = filterTextBox.CaretIndex;
    }
}
```

However I think an AutoCompleteBox should have a bindable property for this info by default.
